### DISCUSSION TITLE
Duplicate Key Violates Unique Constraint

### DISCUSSION THREAD

**Discussion by JSmithOner** [supabase/supabase#8323] [General]
I'm making a bulk upsert with js client library and get:

`{"message":"duplicate key value violates unique constraint \"editor_scripts_pkey\"","code":"23505","details":null,"hint":null}`

everyhting worked fine until now.
I've tried making a `onConflict` column without a chance.
I've seen many posts on the net talking about the database being out of sync, the only problem with the solutions I've found is that they mainly talk about id incrementation but my primary keys ( composite pkeys ) are provided by the client and cannot be incremented

I have 3 primary keys  `project_uid`, `user_uid` and `uid`, one is `text` the other is a `uuid` and the last is `text`

example of one row of data:

```

argstr: ""
channel: 0
col_num: 0
key: 64
max: 256
message_type: 144
min: 0
name:text ""
path: "ozXUwX|8AKQ-P|9Gtpzs|_mtln8"
project_uid: "t2iKq5twmHMKjBspOWtW3"
row_num: 0
script: "this is a script"
type: "type"
uid: "_mtln8"
user_uid: "aacb85bb-4cb9-4bdd-a175-21bc6630a766"

```

also I need to mention even if it's an upsert it happens when row doesn't exist so basically on insertion.

Many thanks in advance.

**Comment by allebd** [comment]
Has this been resolved? I have a similar issue "duplicate key value violates unique constraint \"users_email_partial_key\""

It was working for a long while, then suddenly started throwing the above error for every new user

**Reply by GaryAustin1** [reply]
You probably should start a new question with specifics about what your operation is that gets the error.  The constraint name you show is the exact same as one on auth.users which you should not insert or update directly in general.

**Reply by devsrealm** [reply]
@allebd I had the same error suddenly, it has been working before now, have you been able to fix this?

**Reply by allebd** [reply]
@devsrealm not yet

**Reply by allebd** [reply]
@devsrealm you solution of using a custom SMTP with generateLink worked for me
